diagnostics-extension/src/dpsl.types.ts

```diff
@@ -0,0 +1,258 @@
+ * Response message containing BlockDevice Info
+ */
+export type BlockDeviceInfo = BlockDeviceInfoObject[];
+
+/**
+ * Response message containing StatefulPartition Info
+ */
+export interface StatefulPartitionInfo {
+  availableSpace: bigint;
+  totalSpace: bigint;
+}
+
+/**
+ * Response message containing Timezone Info
+ */
+export interface TimezoneInfo {
+  posix: string;
+  region: string;
+}
+
+// dpsl.diagnostics.* type definitions
+
+/**
+ * |progressPercent| percentage of the routine progress.
+ * |output| accumulated output, like logs.
+ * |status| current status of the routine.
+ * |detail| more detailed status message.
+ * |userMessage| Request user action. Should be localized and displayed to the
+ * user. Note: used in interactive routines only, two possible values are
+ * returned: 'unplug-ac-power' or 'plug-in-ac-power'.
+ */
+export interface RoutineStatus {
+  progressPercent: number;
+  output: string;
+  status: string;
+  statusMessage: string;
+  userMessage: string;
+}
+
+/**
+ * Params object of dpsl.diagnostics.battery.runChargeRoutine()
+ */
+export interface BatteryChargeRoutineParams {
+  lengthSeconds: number;
+  minimumChargePercentRequired: number;
+}
+
+/**
+ * Params object of dpsl.diagnostics.battery.runDischargeRoutine()
+ */
+export interface BatteryDischargeRoutineParams {
+  lengthSeconds: number;
+  maximumDischargePercentAllowed: number;
+}
+
+/**
+ * Params object of dpsl.diagnostics.nvme.runWearLevelRoutine()
+ */
+export interface NvmeWearLevelRoutineParams {
+  wearLevelThreshold: number;
+}
+
+/**
+ * Params object of dpsl.diagnostics.power.{runAcConnectedRoutine(),
+ * runAcDisconnectedRoutine()}
+ */
+export interface PowerAcRoutineParams {
+  expectedPowerType: string;
+}
+
+/**
+ * Params object of dpsl.diagnostics.cpu.{runCacheRoutine(), runStressRoutine(),
+ * runFloatingPointAccuracyRoutine()}
+ */
+export interface CpuRoutineDurationParams {
+  duration: number;
+}
+
+/**
+ * Params object of dpsl.diagnostics.cpu.runPrimeSearchRoutine()
+ */
+export interface CpuPrimeSearchRoutineParams {
+  lengthSeconds: number;
+  maximumNumber: number;
+}
+
+/**
+ * Params object of dpsl.diagnostics.disk.run{Linear/Random}ReadRoutine()
+ */
+export interface DiskReadRoutineParams {
+  lengthSeconds: number;
+  fileSizeMB: number;
+}
+
+// dpsl.system_events.* type definitions
+
+/**
+ * The list of supported system events
+ */
+export enum EventTypes {
+  AC_INSERTED = 'ac-inserted',
+  AC_REMOVED = 'ac-removed',
+  BLUEOOTH_ADAPTER_ADDED = 'bluetooth-adapter-added',
+  BLUEOOTH_ADAPTER_PROPERTY_CHANGED = 'bluetooth-adapter-property-changed',
+  BLUEOOTH_ADAPTER_REMOVED = 'bluetooth-adapter-removed',
+  BLUETOOTH_DEVICE_ADDED = 'bluetooth-device-added',
+  BLUETOOTH_DEVICE_PROPERTY_CHANGED = 'bluetooth-device-property-changed',
+  BLUETOOTH_DEVICE_REMOVED = 'bluetooth-device-removed',
+  LID_CLOSED = 'lid-closed',
+  LID_OPENED = 'lid-opened',
+  OS_RESUME = 'os-resume',
+  OS_SUSPEND = 'os-suspend',
+}
```

Comment:
These are internal definitions in dpsl. As a user of the library, you are going to call very specific event listeners, e.g. dpsl.system_events.bluetooth.addOnAdapterAddedListener(callback). More examples: https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/test/untrusted_browsertest.js?q=dpsl.system_events

Reply:
>/**
     * Starts listening on bluetooth-adapter-added events.
     * @param {function()} callback
     * @public
     */
    addOnAdapterAddedListener(callback) {
      this.eventTarget.addEventListener(
          EVENTS.BLUETOOTH_ADAPTER_ADDED, callback);
    }
    
   Will I receive anything in the callback parameters? If yes, then where can I find their schemas?

Reply:
Callbacks[1] receive no input. It is just state change event, if you need more information, you can call the appropriate API. Example, you receive a callback for bluetooth-adapter-added, then you consult dpsl.telemetry.getBluetoothInfo() and get dpsl.BluetoothInfo[2] as a result.

[1] https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/resources/dpsl/events_service.js

[2] https://source.chromium.org/chromium/chromium/src/+/main:chromeos/components/telemetry_extension_ui/resources/dpsl/dpsl_types.js;drc=043c35d129e66b4ebe45509c4070a39a10fbd06f;l=52

Reply:
Makes sense. Thanks! 👍🏻 I've removed them from code.